backend/src/ee/routes/v1/external-kms-router.ts

```diff
@@ -286,4 +291,89 @@ export const registerExternalKmsRouter = async (server: FastifyZodProvider) => {
       return { externalKms };
     }
   });
+
+  server.route({
+    method: "POST",
+    url: "/fetch-gcp-keys/credential",
+    config: {
+      rateLimit: writeLimit
+    },
+    schema: {
+      body: z.object({
+        region: z.string().trim().min(1),
+        credential: ExternalKmsGcpCredentialSchema
+      }),
+
+      response: {
+        200: z.object({
+          keys: z.array(z.string())
+        })
+      }
+    },
+    onRequest: verifyAuth([AuthMode.JWT, AuthMode.IDENTITY_ACCESS_TOKEN]),
+    handler: async (req) => {
+      const { region, credential } = req.body;
+
+      const results = await server.services.externalKms.fetchGcpKeys({
+        credential,
+        gcpRegion: region,
+        keyName: ""
+      });
+      if (!results) {
+        throw new Error("Failed to validate GCP credential");
+      }
```

Comment:
hmm how would this happen if we're always returning an object

**
<img width="746" alt="image" src="https://github.com/user-attachments/assets/179b69e9-6a5c-4dd8-93a8-73f4fce96130">
**